File: 3154-maximum-value-of-an-ordered-triplet-i/maximum-value-of-an-ordered-triplet-i.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumTripletValue = function(nums) {
    
    let ans = 0;

    for(let i = 0; i < nums.length; i++) {

        for(let j = i + 1; j < nums.length; j++) {

            let t = nums[i] - nums[j] ;

            for(let k = j + 1; k < nums.length; k++) {

                ans = Math.max(( t * nums[k]) , ans) 
            }
        }
    }

    return ans;
};

File: 3154-maximum-value-of-an-ordered-triplet-i/maximum-value-of-an-ordered-triplet-i.cpp
class Solution {
public:
    long long maximumTripletValue(vector<int>& nums) {
        
        long long ans = 0;

        for(int i = 0; i < nums.size(); i++) {

            for(int j = i + 1; j < nums.size(); j++) {

                long long temp = nums[i] - nums[j];

                for(int k = j + 1;k < nums.size(); k++) {

                    ans = max(ans, temp * nums[k]);
                }
            }
        }

        return ans;
    }
};